apps/angular/react-in-angular/src/app/react/post.component.tsx

```diff
@@ -0,0 +1,48 @@
+import { AfterViewInit, Component, ElementRef, EventEmitter, input, OnChanges, Output, ViewChild } from '@angular/core';
+import { createRoot, Root } from 'react-dom/client';
+import React from 'react';
+import ReactPost from './ReactPost';
+
+type Post = { title: string; description: string, pictureLink: string };
+
+
+@Component({
+  standalone: true,
+  selector: 'app-post',
+  template: `
+    <div #react></div>`,
+  styles: ['']
+})
+export class PostComponent implements AfterViewInit, OnChanges {
+  @ViewChild('react', { static: true }) containerRef!: ElementRef;
+
+  post = input<Post | undefined>(undefined);
+  isSelected = input<boolean>(false);
+  @Output() selectPost = new EventEmitter<void>();
+  root?: Root;
+
+  ngOnChanges(): void {
+    this.render();
```

Comment:
Why did you add it here as well?

Reply:
@tomalaforge Are you referring to `root` ?

Reply:
@jdegand no, why do we need to rerender at each change?

Reply:
Well, we might not need it in this case since the posts are not changing.  I think it is typically done to ensure that the Angular component will re-render if there is any change in the React component.  React and Angular have different change detection strategies.   React avoids unnecessary changes (virtual DOM) so there could be a mismatch.

Reply:
Thanks for the explanation. I have never used React.